File: py/leetcode-widestverticalarea.py
class Solution:
    def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
        xs = [c[0] for c in points]

        xs.sort()

        maxgap = -1
        for i in range(len(xs) - 1):
            maxgap = max(xs[i+1]-xs[i], maxgap)
        
        return maxgap

File: cpp/leetcode-widestverticalarea.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int maxWidthOfVerticalArea(vector<vector<int>>& points) {
        vector<int> xs;
        for (vector<int> p:points) {
            xs.push_back(p[0]);
        }

        sort(xs.begin(), xs.end());

        int maxgap = -1000;
        for (int i = 0; i < xs.size() - 1; i ++) {
            maxgap = max(xs[i + 1] - xs[i], maxgap);
        }

        return maxgap;
    }
};